src/pages/workspaces/workspace/workspace-utils.ts

```diff
@@ -0,0 +1,43 @@
+export type CloudProviderType = 'AZURE' | 'GCP'
+export const cloudProviderTypes: Record<CloudProviderType, CloudProviderType> = {
+  AZURE: 'AZURE',
+  GCP: 'GCP'
+}
+export const isCloudProvider = (x: unknown): x is CloudProviderType => {
+  return x as string in cloudProviderTypes
+}
+
+export interface BaseWorkspaceInfo {
```

Comment:
Nit: "Info" seems superfluous. I'd suggest leaving it off and naming this "BaseWorkspace"

Reply:
Hmm... I like this comment but in the other direction. Intention here is to differentiate between two variables often both named `workspace`. I will rename the parent object as `BaseWorkspace` from `BaseWorkspaceWrapper`, but keep `BaseWorkspaceInfo` as is (which appears as a child within a `BaseWorkspace`